Sources/TuistSupport/Extensions/FileManager+Extras.swift

```diff
@@ -5,9 +5,19 @@ extension FileManager {
         subdirectoriesResolvingSymbolicLinks(atNestedPath: nil, basePath: path)
     }
 
-    private func subdirectoriesResolvingSymbolicLinks(atNestedPath nestedPath: String?, basePath: String) -> [String] {
+    private func subdirectoriesResolvingSymbolicLinks(atNestedPath nestedPath: String?, basePath: String, visitedPaths: inout Set<String>) -> [String] {
```

Comment:
Thanks a lot for contributing this fix @haifengkao. I'd add a test that recreates the scenario in a temporary directory, invoke the logic here, and ensure the path that you get back are the expected one. Once we have the test we can merge the fix.